docs/index.bs

```diff
@@ -2982,35 +2991,41 @@ spec: webappsec-referrer-policy; urlPrefix: https://w3c.github.io/webappsec-refe
+              1. Let |requestObject| be a new {{Request}} object associated with |request| and a new associated {{Headers}} object whose [=guard=] is "`immutable`".
+              1. Initialize |e|’s {{Event/type}} attribute to {{fetch!!event}}.
+              1. Initialize |e|’s {{Event/cancelable}} attribute to true.
+              1. Initialize |e|’s {{FetchEvent/request}} attribute to |requestObject|.
+              1. Initialize |e|’s {{FetchEvent/preloadResponse}} to |preloadResponse|.
+              1. Initialize |e|'s {{FetchEvent/clientId}} attribute to |client|'s [=environment/id=].
+              1. If |request| is a <a>non-subresource request</a> and |request|'s [=request/destination=] is not {{RequestDestination/"report"}}, initialize |e|'s {{FetchEvent/resultingClientId}} attribute to |reservedClient|'s [=environment/id=], and to the empty string otherwise.
+              1. If |request| is a <a>navigation request</a>, initialize |e|'s {{FetchEvent/replacesClientId}} attribute to |request|'s [=request/replaces client id=], and to the empty string otherwise.
+              1. Initialize |e|’s {{FetchEvent/handled}} to |eventHandled|.
+              1. <a>Dispatch</a> |e| at |activeWorker|'s [=service worker/global object=].
+              1. Invoke [=Update Service Worker Extended Events Set=] with |activeWorker| and |e|.
+              1. If |e|'s [=FetchEvent/respond-with entered flag=] is set, set |respondWithEntered| to true.
+              1. If |e|'s [=FetchEvent/wait to respond flag=] is set, then:
+                  1. Wait until |e|'s [=FetchEvent/wait to respond flag=] is unset.
+                  1. If |e|'s [=FetchEvent/respond-with error flag=] is set, set |handleFetchFailed| to true.
+                  1. Else, set |response| to |e|'s [=FetchEvent/potential response=].
+              1. If |e|'s <a>canceled flag</a> is set, set |eventCanceled| to true.
+              1. If |fetchInstance| is [=fetch/terminated=], then [=queue a task=] to [=AbortSignal/signal abort=] on |requestObject|'s {{Request/signal}}.
+
+             If |task| is discarded, set |handleFetchFailed| to true.
+
+             The |task| *must* use |activeWorker|'s <a>event loop</a> and the <a>handle fetch task source</a>.
 
       1. Wait for |task| to have executed or for |handleFetchFailed| to be true.
       1. If |shouldSoftUpdate| is true, then [=in parallel=] run the [=Soft Update=] algorithm with |registration|.
-      1. If |respondWithEntered| is false, then return a [=network error=] if |eventCanceled| is true and null otherwise.
-      1. If |handleFetchFailed| is true, then return a [=network error=].
-      1. Return |response|.
+      1. If |respondWithEntered| is false, then:
+          1. If |eventCanceled| is true, then [=reject=] |eventHandled| with a "{{NetworkError}}" {{DOMException}} and return a [=network error=].
+          1. [=resolve=] |eventHandled| and return null.
```

Comment:
```suggestion
          1. [=Resolve=] |eventHandled| and return null.
```

Reply:
[=Resolve=] seems compiled to be linked to 'https://w3c.github.io/FileAPI/#blob-url-resolve' in the output. Shall we drop the link by using simple text like "Resolve |eventHandled| and ...".